### PR TITLE
Figure.rose: Deprecate parameter "columns" to "incols" (remove in v0.6.0)

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

This PR adds the deprecate_parameter decorator for backward-compatibility to Figure.rose().

- [x] Updates tests and examples to use the new parameter name
- [x] Use the deprecate_parameter decorator for backward-compatibility
- [x] Add a test for checking 'columns' backward compatibility

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Write detailed docstrings for all functions/methods.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_rose.py [hunk 1]
```diff
@@ -188,3 +188,32 @@ def test_rose_bools(data_fractures_compilation):
         shift=False,
     )
     return fig
+
+
+@pytest.mark.mpl_image_compare(filename="test_rose_bools.png")
+def test_rose_deprecate_columns_to_incols(data_fractures_compilation):
+    """
+    Make sure that the old parameter "columns" is supported and it reports an
```
Comment (seisman) [inline pygmt/tests/test_rose.py:196]:
```suggestion
    Make sure that the old parameter "columns" is supported and it reports a
```

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
For unknown reasons, the `test_rose_data_file` test fails and the `test_rose_no_sectors` hangs (likely waiting for input file).
```
../pygmt/tests/test_rose.py::test_rose_data_file FAILED                                                                         [ 84%]
../pygmt/tests/test_rose.py::test_rose_2d_array_single PASSED                                                                   [ 84%]
../pygmt/tests/test_rose.py::test_rose_2d_array_multiple PASSED                                                                 [ 84%]
../pygmt/tests/test_rose.py::test_rose_plot_data_using_cpt PASSED                                                               [ 85%]
../pygmt/tests/test_rose.py::test_rose_plot_with_transparency FAILED                                                            [ 85%]
../pygmt/tests/test_rose.py::test_rose_no_sectors
```

File: pygmt/src/rose.py [hunk 2]
```diff
@@ -3,10 +3,17 @@
...
+    build_arg_string,
+    deprecate_parameter,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
 
 
 @fmt_docstring
+@deprecate_parameter("columns", "incols", "v0.4.0", remove_version="v0.6.0")
```
Comment (seisman) [inline pygmt/src/rose.py:16]: @michaelgrund You also need to change `i="columns"` to `i="incols"` at line 32.

Comment (michaelgrund) [reply pygmt/src/rose.py:16]: Sure, thanks for mentioning @seisman 😉.

**Pull request merged**